linchpin/__init__.py

```diff
@@ -569,6 +572,7 @@ def do_action(self, provision_data, action='up', run_id=None, tx_id=None):
                    to perform an idempotent reprovision, or destroy provisioned
                    resources.
         """
+        self.flags = flags
```

Comment:
You could get the flags by doing something like `flags.append(ctx.get_cfg('hooks', 'no_hooks', default='False'))` for each of the flags, or have a way to load them from the config/cli.